src/Mod/TechDraw/App/HatchLine.cpp

```diff
@@ -186,40 +176,41 @@ Base::Vector3d LineSet::getPatternStartPoint(TechDraw::BaseGeomPtr g, double &of
         offset = 0.0;
-    } else {
-        //find a point where pattern repeats within g
-        double patsStartOrg = lenStartOrg/patternLength;                   //# pattern repeats from lineOrigin to start
-        double patsEndOrg = lenEndOrg/patternLength;
-        if (lenStartOrg < lenEndOrg) {                                  //origin is before start
-            double c = ceil(patsStartOrg);
-            if (c <= patsEndOrg) {                                      //c is an integer pattern count in [patsStartOrg,patsEndOrg]
-                result = lineOrigin + c*patternLength*getUnitDir();
-                offset = 0.0;
-            } else {
+        return lineOrigin;
+    }
+
+    //find a point where pattern repeats within g
+    double patternLength = scale * getPatternLength();
+    double patsStartOrg = lenStartOrg/patternLength;                   //# pattern repeats from lineOrigin to start
+    double patsEndOrg = lenEndOrg/patternLength;
+    if (lenStartOrg < lenEndOrg) {                                  //origin is before start
+        double c = ceil(patsStartOrg);
+        if (c <= patsEndOrg) {                                      //c is an integer pattern count in [patsStartOrg,patsEndOrg]
+            offset = 0.0;
+            return lineOrigin + c*patternLength*getUnitDir();
+        } else {
 //                //ugly case - partial pattern
-                result = gStart;
-                offset = patsStartOrg - (int)patsStartOrg;        //fraction of a patternLength
-                offset = offset * patternLength;
-            }
-        } else if (lenStartOrg > lenEndOrg) {                          //origin is after end
-            double c = ceil(patsEndOrg);
-            if (c <= patsStartOrg) {                      //c is an integer pattern count in [patsStartOrg,patsEndOrg]
-                result = lineOrigin - c*patternLength*getUnitDir();
-                offset = 0.0;
-            } else {
-                result = gStart;
-                offset = ceil(patsStartOrg) - patsStartOrg;      //fraction of a patternLength patstartorg to repeat point
-                offset = offset * patternLength;
-            }
+            offset = patsStartOrg - (int)patsStartOrg;        //fraction of a patternLength
+            offset = offset * patternLength;
+            return gStart;
+        }
+    } else if (lenStartOrg > lenEndOrg) {                          //origin is after end
+        double c = ceil(patsEndOrg);
+        if (c <= patsStartOrg) {                      //c is an integer pattern count in [patsStartOrg,patsEndOrg]
+            offset = 0.0;
+            return lineOrigin - c*patternLength*getUnitDir();
         } else {
-            Base::Console().Log("ERROR - HL::getPatternStart - something has gone wrong!\n");
+            offset = ceil(patsStartOrg) - patsStartOrg;      //fraction of a patternLength patstartorg to repeat point
+            offset = offset * patternLength;
```

Comment:
Won't this overwrite offset?

Reply:
It is supposed to overwrite offset.  I guess it should really be returning std::pair<Base::Vector3d, double>.